Modules/_io/iobase.c

```diff
@@ -946,9 +952,14 @@ _io__RawIOBase_read_impl(PyObject *self, Py_ssize_t n)
         return NULL;
     }
 
-    res = PyBytes_FromStringAndSize(PyByteArray_AsString(b), n);
-    Py_DECREF(b);
-    return res;
+    if (n != PyBytes_GET_SIZE(b)) {
+        if (_PyBytes_Resize(&b, n) < 0) {
+            Py_DECREF(b);
+            return NULL;
+        }
+    }
+
+    return b;
```

Comment:
```suggestion
    if (_PyBytes_Resize(&b, n) < 0) {
        Py_DECREF(b);
        return NULL;
    }
    return b;
```

`_PyBytes_Resize` has a fast == sizes.